### PR TITLE
Feature/docs 1524 copy blocking page

### PR DESCRIPTION
**Pull request by MaximBashurov**

### CODE REVIEW

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ By default, the response code 403 and default NGINX blocking page are returned t
 
 You can configure the blocking page and error code passing the following parameters in the `wallarm_block_page` NGINX directive:
 
-* Path to the HTM or HTML file of the blocking page. You can specify the path either to a custom blocking page or the [default blocking page](#customizing-default-blocking-page) provided by Wallarm.
+* Path to the HTM or HTML file of the blocking page. You can specify the path either to a custom blocking page or the [sample blocking page](#customizing-sample-blocking-page) provided by Wallarm.
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:24]: The task is to prevent the user from using the sample blocking page as is. But in this list we say that the user can specify the path to the sample blocking page

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 2]
```diff
@@ -85,20 +85,276 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
...
 * Adding your company logo – by default, no logo is presented on the page.
 * Adding your company support email – by default, no email links are used and the `contact us` phrase is the simple text without any link.
 * Changing any other HTML elements or adding your own.
 
-!!! info "Default vs custom blocking page"
-    Instead of modifying the default page provided by Wallarm, you can create your own custom one and then set it to be used by specifying path to it via the [wallarm_block_page](#nginx-directive-wallarm_block_page) or the [wallarm_block_page_add_dynamic_path](#nginx-directive-wallarm_block_page_add_dynamic_path) directive.
+!!! info "Sample vs custom blocking page"
+    Instead of modifying the sample page provided by Wallarm, you can create your own custom one and then set it to be used by specifying path to it via the [wallarm_block_page](#nginx-directive-wallarm_block_page) or the [wallarm_block_page_add_dynamic_path](#nginx-directive-wallarm_block_page_add_dynamic_path) directive.
+
+### General procedure
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:103]: Maybe we can merge this information section with the information provided in the "configuration examples" block? unless, it looks like the duplicated info

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md (hunk 1 above)
Comment (MaximBashurov) [reply docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:24]: > The task is to prevent the user from using the sample blocking page as is.

This was not the task, we still have the following text:

> You can use the sample page without customization

The task was to protect modifications of the sample page IF any.

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md (hunk 2 above)
Comment (MaximBashurov) [reply docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:103]: As I wrote in the previous comment, users can use the sample page "as is". So I think that is quite appropriate that the instruction "Examples" show how to do that - users only need additional steps if they want to MODIFY the sample page.

I also see no problem in some level of duplicating - even before our changes we had "To apply the settings to the Docker" and further steps repeated 4 times and that was quite not a problem - the document was very clear and usable.

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 3]
```diff
@@ -85,22 +83,278 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
...
+You can use the sample page as a start point for your customization enhancing it by:
 
 * Adding your company logo – by default, no logo is presented on the page.
 * Adding your company support email – by default, no email links are used and the `contact us` phrase is the simple text without any link.
 * Changing any other HTML elements or adding your own.
 
-!!! info "Default vs custom blocking page"
-    Instead of modifying the default page provided by Wallarm, you can create your own custom one and then set it to be used by specifying path to it via the [wallarm_block_page](#nginx-directive-wallarm_block_page) or the [wallarm_block_page_add_dynamic_path](#nginx-directive-wallarm_block_page_add_dynamic_path) directive.
+!!! info "Custom blocking page variants"
+    Instead of modifying the sample page provided by Wallarm, you can create custom page from scratch.
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:99]: missed article before "custom page"

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 4]
```diff
@@ -85,22 +83,278 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
...
+
+### General procedure
+
+If you modify the sample page itself, your modifications may be lost on Wallarm components update. Therefore it is recommended to copy sample page, give it new name and only then modify. Act depending on your installation type as described in the sections below.
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:103]: missed articles and punctuation. please check the document with grammarly

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 5]
```diff
@@ -85,22 +83,278 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
...
+
+**<a name="copy"></a>Sample page for copying**
+
+You can make copy of the `/usr/share/nginx/html/wallarm_blocked.html` located in the environment where your filtering node is installed. Alternatively, copy the code below and save it as your new file:
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:107]: it looks like this block purpose is different from all the following blocks. this block provides the sample page code, but other blocks - the configuration methods. my suggestion is to move this block to the sample page overview section. what do you think?

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 6]
```diff
@@ -85,22 +83,278 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
...
+You can make copy of the `/usr/share/nginx/html/wallarm_blocked.html` under a new name wherever you want (NGINX should have read permission there) including the same folder.
+
+**Docker container**
+
+To modify sample blocking page or provide your own custom from scratch, you can use Docker's [bind mount](https://docs.docker.com/storage/bind-mounts/) functionality. When using it, your page and NGINX configuration file from your host machine are copied to the container and then referenced with the originals, so that if you change files on the host machine, their copies will be synchronized and vise versa.
+
+Therefore, to modify sample blocking page or provide your own, do the following:
+
+1. Before first run, [prepare](#copy) your modified `wallarm_blocked-renamed.html`.
+1. Prepare NGINX configuration file with the path to your blocking page. See [Configuration examples →](#configuration-examples)
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:334]: you can give a deeper link right to the described conf example

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 7]
```diff
@@ -85,22 +83,278 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
...
+1. Run the container [mounting](../installation-docker-en.md#run-the-container-mounting-the-configuration-file) the prepared blocking page and configuration file.
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:335]: broken link

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 8]
```diff
@@ -85,22 +83,278 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
...
+1. If you need later to update your blocking page in a running container, on the host machine, change the referenced `wallarm_blocked-renamed.html` then restart NGINX in the container.
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:336]: this example says the mounted file is updated automatically but the example in the "configuration examples" is with the `-v` option. this option does not allow the automatic update of mounted files

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 9]
```diff
@@ -134,13 +388,13 @@ The `type` parameter of the `wallarm_block_page` directive is explicitly specifi
...
+* [Modified](#customizing-sample-blocking-page) sample Wallarm blocking page `/usr/share/nginx/html/wallarm_blocked-renamed.html` and the error code 445 returned if the request is blocked by the filtering node in the blocking or safe blocking mode.
 * Custom blocking page `/usr/share/nginx/html/block.html` and the error code 445 returned if the request is originated from any blacklisted IP address.
 
 #### NGINX configuration file
 
 ```bash
-wallarm_block_page &/usr/share/nginx/html/wallarm_blocked.html response_code=445 type=attack;
+wallarm_block_page &/usr/share/nginx/html/wallarm_blocked-renamed.html response_code=445 type=attack;
 wallarm_block_page &/usr/share/nginx/html/block.html response_code=445 type=acl_ip,acl_source;
 ```
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:399]: below is the phrase "If configuring the custom blocking page, this page should also be mounted to the container". is not the condition is extra?

Comment (AnastasiaTWW) [reply docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:399]: the same for the "Variable and error code[¶](https://deploy-preview-503--pensive-dubinsky-5f7a00.netlify.app/admin-en/configuration-guides/configure-block-page-and-code/#variable-and-error-code)" section

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 10]
```diff
@@ -134,13 +388,13 @@ The `type` parameter of the `wallarm_block_page` directive is explicitly specifi
...
```
[the remark this answers is not in the document]
Comment (AnastasiaTWW) [reply docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:400]: the same for the "Variable and error code[¶](https://deploy-preview-503--pensive-dubinsky-5f7a00.netlify.app/admin-en/configuration-guides/configure-block-page-and-code/#variable-and-error-code)" section

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 11]
```diff
@@ -85,22 +83,278 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
...
+
+You can make copy of the `/usr/share/nginx/html/wallarm_blocked.html` under a new name wherever you want (NGINX should have read permission there) including the same folder.
+
+**Docker container**
+
+To modify sample blocking page or provide your own custom from scratch, you can use Docker's [bind mount](https://docs.docker.com/storage/bind-mounts/) functionality. When using it, your page and NGINX configuration file from your host machine are copied to the container and then referenced with the originals, so that if you change files on the host machine, their copies will be synchronized and vise versa.
+
+Therefore, to modify sample blocking page or provide your own, do the following:
+
+1. Before first run, [prepare](#copy) your modified `wallarm_blocked-renamed.html`.
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:333]: why "wallarm_blocked-renamed.html" but not "wallarm_blocked_renamed.html"?

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 12]
```diff
@@ -85,22 +83,278 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
 
 The directive  can be set inside the `http` block of the NGINX configuration file.
 
-## Customizing default blocking page
+## Customizing sample blocking page
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:86]: if there are links to the section with the old name, please update them

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md (hunk 5 above)
Comment (MaximBashurov) [reply docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:107]: Disagree: we have 2 lines above this: "it is recommended to copy the sample page, give it a new name, and only then modify" and then... here you are! take it and copy.

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md (hunk 6 above)
Comment (MaximBashurov) [reply docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:334]: There are 5 different "NGINX configuration file" examples under "Configuration examples" - any of them can be reviewed and thus I cannot see how the deeper link could be useful.

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md (hunk 8 above)
Comment (MaximBashurov) [reply docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:336]: I used:
docker run -d -e DEPLOY_USER='mbashurov+firsttest@wallarm.com' -e DEPLOY_PASSWORD='9ol.0p;/B1' -v /configs/default:/etc/nginx/sites-enabled/default -p 80:80 wallarm/node:3.6.0-1
...with `-v` option and then tested - files were updated in both directions so this was a [bind mount](https://docs.docker.com/storage/bind-mounts/)

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md (hunk 6 above)
Comment (AnastasiaTWW) [reply docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:334]: there are only 2 suitable examples in this doc, not all 5 examples describe the configuration of customized blocking page return. the first one of them seems to be more suitable for this case.

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 13]
```diff
@@ -486,28 +486,28 @@ This configuration is returned to the client if the request is originated from t
...
   "~Chrome"   &/usr/share/nginx/html/block_page_chrome.html;
-  default     &/usr/share/nginx/html/wallarm_blocked-renamed.html;
+  default     &/usr/share/nginx/html/wallarm_blocked_renamed.html;
 }
 
 wallarm_block_page $block_page response_code=445 type=acl_ip;
 ```
 
-* To apply the settings to the Docker container, the NGINX configuration file with appropriate settings should be mounted to the container. If configuring the custom blocking page, this page should also be mounted to the container. [Running the container mounting the configuration file →](../installation-docker-en.md#run-the-container-mounting-the-configuration-file)
+* To apply the settings to the Docker container, the NGINX configuration file with appropriate settings should be mounted to the container. [Running the container mounting the configuration file →](../installation-docker-en.md#run-the-container-mounting-the-configuration-file)
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:500]: the page also should be mounted to the container

File: docs/latest/updating-migrating/docker-container.md [hunk 14]
```diff
@@ -62,13 +62,17 @@ There are the following deprecated configuration options:
...
+In the new node version, the Wallarm sample blocking page has [been changed](what-is-new.md#when-upgrading-node-34). The logo and support email on the page are now empty by default.
 
-If the Docker container 2.18 or lower is configured to return the `&/usr/share/nginx/html/wallarm_blocked.html` page to blocked requests, change this configuration as follows:
+If your Docker container was configured to return some specific page in response to the blocked requests, transfer this configuration as follows:
 
-1. Copy the blocking page file contents from the new Docker image file system.
-2. [Customize](..//admin-en/configuration-guides/configure-block-page-and-code.md#customizing-default-blocking-page) the copied page.
-3. [Mount](../admin-en/configuration-guides/configure-block-page-and-code.md#path-to-the-htm-or-html-file-with-the-blocking-page-and-error-code) the customized page and the NGINX configuration file to a new Docker container in the next step.
+1. Optionally, [copy and customize](..//admin-en/configuration-guides/configure-block-page-and-code.md#customizing-sample-blocking-page) the new version of a sample page.
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/docker-container.md:69]: if the first step is optional, the second step is optional too

Comment (AnastasiaTWW) [reply docs/latest/updating-migrating/docker-container.md:69]: + "optionally" is extra here since you already provided the user with the conditions to perform these steps

File: docs/latest/updating-migrating/docker-container.md [hunk 15]
```diff
@@ -62,13 +62,17 @@ There are the following deprecated configuration options:
 
 ## Step 4: Update the Wallarm blocking page (if upgrading NGINX-based image)
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/docker-container.md:63]: Is "Wallarm blocking page" here is suitable? not sample page?

File: docs/latest/updating-migrating/cloud-image.md [hunk 16]
```diff
@@ -52,9 +52,11 @@ To upgrade the node 2.18 or lower, please use the [different instructions](older
     * `wallarm_global_trainingset_path` → [`wallarm_protondb_path`](../admin-en/configure-parameters-en.md#wallarm_protondb_path)
 
     We only changed the names of the directives, their logic remains the same. Directives with former names will be deprecated soon, so you are recommended to rename them before.
-1. If the page `&/usr/share/nginx/html/wallarm_blocked.html` is returned to blocked requests, [copy and customize](../admin-en/configuration-guides/configure-block-page-and-code.md#customizing-default-blocking-page) its new version.
+1. Optionally, update the page returned in response to the blocked requests by [copying or customizing](../admin-en/configuration-guides/configure-block-page-and-code.md#customizing-sample-blocking-page) the new version of a sample page.
```
[the remark this answers is not in the document]
Comment (AnastasiaTWW) [reply docs/latest/updating-migrating/cloud-image.md:55]: the same for the changes in other instructions

File: docs/latest/updating-migrating/nginx-modules.md [hunk 17]
```diff
@@ -136,9 +136,11 @@ Execute the following command to upgrade the filtering node and postanalytics mo
 
 ## Step 4: Update the Wallarm blocking page
 
-In new node versions, the Wallarm blocking page has [been changed](what-is-new.md#when-upgrading-node-34). The logo and support email on the page are now empty by default.
+In the new node version, the Wallarm sample blocking page has [been changed](what-is-new.md#when-upgrading-node-34). The logo and support email on the page are now empty by default.
 
-If the page `&/usr/share/nginx/html/wallarm_blocked.html` is returned to blocked requests, [copy and customize](../admin-en/configuration-guides/configure-block-page-and-code.md#customizing-default-blocking-page) its new version.
+If some specific page was configured to be returned in response to the blocked requests, optionally [copy and customize](../admin-en/configuration-guides/configure-block-page-and-code.md#customizing-sample-blocking-page) the new version of a sample page.
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/nginx-modules.md:141]: the same about specific page

File: docs/latest/updating-migrating/nginx-modules.md [hunk 18]
```diff
@@ -136,9 +136,11 @@ Execute the following command to upgrade the filtering node and postanalytics mo
...
+
+    Note that you can also continue using your previous custom page or build a new one from scratch, not using the sample.
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/nginx-modules.md:143]: extra indent?

File: docs/latest/updating-migrating/ingress-controller.md [hunk 19]
```diff
@@ -36,9 +36,11 @@ To add or update the parameters of the upgraded Helm chart, use one more command
 1. If there is the annotation `nginx.ingress.kubernetes.io/wallarm-instance` explicitly passed to the Helm chart, rename it to `nginx.ingress.kubernetes.io/wallarm-application`.
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/ingress-controller.md:36]: should not we change the "older-versions/*" instructions in the same way?

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 20]
```diff
@@ -85,22 +83,278 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
...
 * Changing any other HTML elements or adding your own.
 
-!!! info "Default vs custom blocking page"
-    Instead of modifying the default page provided by Wallarm, you can create your own custom one and then set it to be used by specifying path to it via the [wallarm_block_page](#nginx-directive-wallarm_block_page) or the [wallarm_block_page_add_dynamic_path](#nginx-directive-wallarm_block_page_add_dynamic_path) directive.
+!!! info "Custom blocking page variants"
+    Instead of modifying the sample page provided by Wallarm, you can create a custom page from scratch.
+
+### General procedure
+
+If you modify the sample page itself, your modifications may be lost on Wallarm components update. Therefore, it is recommended to copy the sample page, give it a new name, and only then modify. Act depending on your installation type as described in the sections below.
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:103]: modify IT?

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md (hunk 4 above)
Comment (AnastasiaTWW) [reply docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md]: the are still some open questions

File: docs/latest/updating-migrating/cloud-image.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (MaximBashurov) [reply docs/latest/updating-migrating/cloud-image.md:55]: Let's discuss.

File: docs/latest/updating-migrating/docker-container.md (hunk 14 above)
Comment (MaximBashurov) [reply docs/latest/updating-migrating/docker-container.md:69]: Let's discuss.

File: docs/latest/updating-migrating/docker-container.md (hunk 15 above)
Comment (MaximBashurov) [reply docs/latest/updating-migrating/docker-container.md:63]: Yes for the header, because we modify the blocking page which may be custom and not limited to sample or modified sample. However, inside the section, I changed:
"In new node versions, the Wallarm blocking page has [been changed]" →
"In new node versions, the Wallarm sample blocking page has [been changed]"

File: docs/latest/updating-migrating/docker-container.md [hunk 21]
```diff
@@ -62,13 +62,17 @@ There are the following deprecated configuration options:
...
 
-In new node versions, the Wallarm blocking page has [been changed](what-is-new.md#when-upgrading-node-34). The logo and support email on the page are now empty by default.
+In the new node version, the Wallarm sample blocking page has [been changed](what-is-new.md#when-upgrading-node-34). The logo and support email on the page are now empty by default.
 
-If the Docker container 2.18 or lower is configured to return the `&/usr/share/nginx/html/wallarm_blocked.html` page to blocked requests, change this configuration as follows:
+If your Docker container was configured to return some specific page in response to the blocked requests, transfer this configuration as follows:
```
[the remark this answers is not in the document]
Comment (MaximBashurov) [reply docs/latest/updating-migrating/docker-container.md:67]: In the previous version of the container, any page could be used, not only the sample of the modified sample.

File: docs/latest/updating-migrating/ingress-controller.md (hunk 19 above)
Comment (MaximBashurov) [reply docs/latest/updating-migrating/ingress-controller.md:36]: Let's discuss.

File: docs/latest/updating-migrating/cloud-image.md [hunk 22]
```diff
@@ -54,7 +54,8 @@ To upgrade the node 2.18 or lower, please use the [different instructions](older
     We only changed the names of the directives, their logic remains the same. Directives with former names will be deprecated soon, so you are recommended to rename them before.
 1. If the page `&/usr/share/nginx/html/wallarm_blocked.html` is returned to blocked requests, [copy and customize](../admin-en/configuration-guides/configure-block-page-and-code.md#customizing-default-blocking-page) its new version.
 
-      In new node versions, the Wallarm blocking page has [been changed](what-is-new.md#when-upgrading-node-34). The logo and support email on the page are now empty by default.
+      In the new node version, the sample blocking page has [been changed](what-is-new.md#when-upgrading-node-34). The logo and support email on the page are now empty by default.
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/cloud-image.md:57]: not spread to older-versions

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 23]
```diff
@@ -322,22 +322,22 @@ You can make copy of the `/usr/share/nginx/html/wallarm_blocked.html` located in
...
+To modify the sample blocking page or provide your own custom from scratch, you can use Docker's [bind mount](https://docs.docker.com/storage/bind-mounts/) functionality. When using it, your page and NGINX configuration file from your host machine are copied to the container and then referenced with the originals, so that if you change files on the host machine, their copies will be synchronized and vice versa.
 
-Therefore, to modify sample blocking page or provide your own, do the following:
+Therefore, to modify the sample blocking page or provide your own, do the following:
 
-1. Before first run, [prepare](#copy) your modified `wallarm_blocked_renamed.html`.
-1. Prepare NGINX configuration file with the path to your blocking page. See [Configuration examples →](#configuration-examples)
+1. Before the first run, [prepare](#copy) your modified `wallarm_blocked_renamed.html`.
+1. Prepare NGINX configuration file with the path to your blocking page. See [configuration example](#path-to-the-htm-or-html-file-with-the-blocking-page-and-error-code) below.
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:334]: what is "below" here for? you already provided the user with the link

File: docs/latest/updating-migrating/cloud-image.md [hunk 24]
```diff
@@ -52,11 +52,10 @@ To upgrade the node 2.18 or lower, please use the [different instructions](older
     * `wallarm_global_trainingset_path` → [`wallarm_protondb_path`](../admin-en/configure-parameters-en.md#wallarm_protondb_path)
 
     We only changed the names of the directives, their logic remains the same. Directives with former names will be deprecated soon, so you are recommended to rename them before.
-1. Optionally, update the page returned in response to the blocked requests by [copying and customizing](../admin-en/configuration-guides/configure-block-page-and-code.md#customizing-sample-blocking-page) the new version of a sample page.
+1. If the page `&/usr/share/nginx/html/wallarm_blocked.html` is returned to blocked requests, [copy and customize](../admin-en/configuration-guides/configure-block-page-and-code.md#customizing-default-blocking-page) its new version.
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/cloud-image.md:55]: #customizing-default-blocking-page - why default?

Comment (AnastasiaTWW) [reply docs/latest/updating-migrating/cloud-image.md:55]: it is not the only one place

File: docs/latest/updating-migrating/docker-container.md [hunk 25]
```diff
@@ -62,13 +62,13 @@ There are the following deprecated configuration options:
...
+In the new node version, the Wallarm sample blocking page has [been changed](what-is-new.md#when-upgrading-node-34). The logo and support email on the page are now empty by default.
 
-If the Docker container 2.18 or lower is configured to return the `&/usr/share/nginx/html/wallarm_blocked.html` page to blocked requests, change this configuration as follows:
+If your Docker container was configured to return the `&/usr/share/nginx/html/wallarm_blocked.html` page in response to the blocked requests, transfer this configuration as follows:
 
-1. Copy the blocking page file contents from the new Docker image file system.
-2. [Customize](..//admin-en/configuration-guides/configure-block-page-and-code.md#customizing-default-blocking-page) the copied page.
-3. [Mount](../admin-en/configuration-guides/configure-block-page-and-code.md#path-to-the-htm-or-html-file-with-the-blocking-page-and-error-code) the customized page and the NGINX configuration file to a new Docker container in the next step.
+1. [Copy and customize](..//admin-en/configuration-guides/configure-block-page-and-code.md#customizing-sample-blocking-page) the new version of a sample page.
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/docker-container.md:69]: please get rid of double "//" in the link

File: docs/latest/updating-migrating/docker-container.md [hunk 26]
```diff
@@ -62,13 +62,13 @@ There are the following deprecated configuration options:
...
+
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/docker-container.md:70]: what is this extra line here for?

File: docs/latest/updating-migrating/older-versions/docker-container.md [hunk 27]
```diff
@@ -77,13 +77,13 @@ There are the following deprecated configuration options:
...
+In new node version, the Wallarm sample blocking page has [been changed](what-is-new.md#new-blocking-page). The logo and support email on the page are now empty by default.
 
-If the Docker container 2.18 or lower is configured to return the `&/usr/share/nginx/html/wallarm_blocked.html` page to blocked requests, change this configuration as follows:
+If the Docker container 2.18 or lower was configured to return the `&/usr/share/nginx/html/wallarm_blocked.html` page to blocked requests, change this configuration as follows:
 
-1. Copy the blocking page file contents from the new Docker image file system.
-2. [Customize](../../admin-en/configuration-guides/configure-block-page-and-code.md#customizing-default-blocking-page) the copied page.
-3. [Mount](../../admin-en/configuration-guides/configure-block-page-and-code.md#path-to-the-htm-or-html-file-with-the-blocking-page-and-error-code) the customized page and the NGINX configuration file to a new Docker container in the next step.
+1. [Copy and customize](../../admin-en/configuration-guides/configure-block-page-and-code.md#customizing-sample-blocking-page) the new version of a sample page.
+
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/docker-container.md:85]: extra line

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 28]
```diff
@@ -39,18 +39,16 @@ The `wallarm_block_page` directive accepts the listed parameters in the followin
...
-    Wallarm provides the default blocking page. To use this page, please specify the path `&/usr/share/nginx/html/wallarm_blocked.html` in the directive value.
+    Wallarm provides the sample blocking page `&/usr/share/nginx/html/wallarm_blocked.html`. You can use this page as a start point for your [customization](#customizing-sample-blocking-page).
 
     !!! warning "Important information for Debian and CentOS users"
         If you use an NGINX version lower than 1.11 installed from [CentOS/Debian](../../waf-installation/nginx/dynamic-module-from-distr.md) repositories, you should remove the `request_id` variable from the page code to display the dynamic blocking page correctly:
         ```
         UUID ${request_id}
         ```
 
-        This applies to both `wallarm_blocked.html` and to the custom block page.
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:52]: why did you delete this sentence?

File: docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md [hunk 29]
```diff
@@ -85,22 +83,278 @@ The directive `wallarm_block_page_add_dynamic_path` is used to initialize the bl
...
+
+To modify the sample blocking page or provide your own, do the following:
+
+1. [Prepare](#copy) your modified `wallarm_blocked_renamed.html`.
+1. [Create ConfigMap from the file](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-configmaps-from-files) `wallarm_blocked_renamed.html`.
+1. Mount created ConfigMap to the pod with Wallarm Ingress controller. For this, please update the Deployment object relevant for Wallarm Ingress controller following the [instructions](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#populate-a-volume-with-data-stored-in-a-configmap).
+
+    !!! info "Directory for mounted ConfigMap"
+        Existing files in the directory used to mount ConfigMap will be deleted.
+
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configuration-guides/configure-block-page-and-code.md:348]: extra line
